File: data/EXP01/python/2420/2420.output_2.py
# @lc app=leetcode id=2420 lang=python3
from typing import List

class Solution:
    def goodIndices(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        result = []
        
        for i in range(k, n-k):
            left_subarray = nums[i-k:i]  # Elements just before the index i
            right_subarray = nums[i+1:i+k+1]  # Elements just after the index i
            
            if left_subarray == sorted(left_subarray, reverse=True) and right_subarray == sorted(right_subarray):
                result.append(i)
        
        return result

File: data/EXP01/golang_no_system_var/2420/2420.output_2.go
func goodIndices(nums []int, k int) []int {
    result := make([]int, 0)

    for i := k; i < len(nums)-k; i++ {
        nonIncreasing := true
        for j := i-1; j >= i-k; j-- {
            if nums[j] < nums[j+1] {
                nonIncreasing = false
                break
            }
        }

        nonDecreasing := true
        for j := i+1; j <= i+k; j++ {
            if nums[j] < nums[j-1] {
                nonDecreasing = false
                break
            }
        }

        if nonIncreasing && nonDecreasing {
            result = append(result, i)
        }
    }

    return result
}
